developersample/src/main/java/com/microsoft/identity/client/developersample/AuthUtil.java

```diff
@@ -0,0 +1,215 @@
+     * providing the component name will help separate the logs from application and the logs from the sdk running inside of
+     * the apps.
+     * To set component name:
+     * {@link PublicClientApplication#setComponent(String)}
+     * <p>
+     * For the {@link AuthenticationCallback}, MSAL exposes three results 1) Success, which contains the {@link AuthenticationResult} 2) Failure case,
+     * which contains {@link MsalException} and 3) Cancel, specifically for user canceling the flow.
+     * <p>
+     * For the failure case, MSAL exposes three sub exceptions:
+     * 1) {@link MsalClientException}, which is specifically for the exceptions running inside the client app itself, could be no active network,
+     * Json parsing failure, etc.
+     * 2) {@link MsalServiceException}, which is the error that the sdk gets back when communicating to the service, could be oauth2 errors, socket timout
+     * or 500/503/504. For oauth2 erros, MSAL returns back the exact error that server returns back to the sdk.
+     * 3) {@link MsalUiRequiredException}, which means that UI is required.
+     */
+    public AuthUtil(final Activity activity) {
+        if (mApplication == null) {
+            mApplication = new PublicClientApplication(activity.getApplicationContext(), CLIENT_ID);
+        }
+        mActivity = activity;
+    }
+
+    // Get the number of users signed into the app. In the context of the current app this method can return 1 or 0
+    int getUserCount() {
+        getUsers();
+        if (mUsers == null) {
+            return 0;
+        }
+
+        return mUsers.size();
+    }
+
+    // Do an interactive login request
+    // Task is an handle to MainActivity which has overrriden useAccessToken and onRequestFailure
+    // On acquireToken success, useAccessToken is called with the AccessToken and on failure onRequestFailure is called
+    // with the exception
+    void doAcquireToken(final AuthenticatedTask task) {
+        // Interactive request must not be called from more than one thread at a time.
+        synchronized (LOCK) {
+            AuthCallback callback = new AuthCallback(task, true);
+            mApplication.acquireToken(mActivity, SCOPES, callback);
+        }
+    }
+
+    // Do an silent login request
+    // Task is an handle to MainActivity which has overrriden useAccessToken and onRequestFailure
+    // On acquireToken success, useAccessToken is called with the AccessToken and on failure onRequestFailure is called
+    // with the exception
+    void doAcquireTokenSilent(final AuthenticatedTask task) {
+        AuthCallback callback = new AuthCallback(task, false);
+        mApplication.acquireTokenSilentAsync(SCOPES, mUsers.get(0), callback);
+    }
+
+    // Remove the user from the list of users being tracked by the application. This does not clear the cookies
+    void doSignout() {
+        if (getUserCount() > 0) {
+            mApplication.remove(mUsers.get(0));
```

Comment:
there will never be multiple users

---
In reply to: [114895844](https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/165#discussion_r114895844) [](ancestors = 114895844)